### PR TITLE
Add SPM Support

### PR DESCRIPTION
**Pull request by nacho4d**
- **Please note this is a personal contribution. I do NOT represent my client, nor my company. I am contributing to this SDK because I know Treasure-Data SDK is quite famous and it would be great to have SPM support.**

## Changes

- Add Swift Package Manager support
    - I did 0 changes to source code. I only separated public/internal/public files and folders so SPM can build an equivalent binary (without exposing more than needed, without exposing less than needed) as current build.
- I added a simple GitHub action that builds using SPM to assure it is not broken in the future.
- I changed IPHONEOS_DEPLOYMENT_TARGET from 5.0 to 9.0 because latests Xcode requires it.

### CODE REVIEW

**Comment by nacho4d** [comment]
Hello Treasure Data team.

I am contributing SPM support to KeenClient so it can work with TreasureData SDK https://github.com/treasure-data/td-ios-sdk/pull/106 .

Could you please review this when you have some time? 

Kind regards

**Review by tung-vu-td — COMMENTED** [review]
Thanks @nacho4d so much for your PR! We really appreciate it.

I added a couple of comments for PR. If you have time, feel free to make the change. Otherwise, I can merge this PR as is next week and make couple of changes myself.

**Comment by nacho4d** [comment]
@tung-vu-td 
Thank for looking at this. I have addressed the comments :)

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import PackageDescription
+
+let package = Package(
+    name: "KeenClientTD",
+    platforms: [
+        .macOS(.v10_10),
+        .iOS(.v11),
+        .tvOS(.v9),
+        .watchOS(.v2)
```
[the remark this answers is not in the document]
Comment (nacho4d) [reply Package.swift]: Thank you. I have removed watchOS

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            sources: [
+                "KeenClient/",
+                "KeenClientInternal/",
+                "Library/sqlite-amalgamation/",
+            ],
+            publicHeadersPath: "KeenClient",
+            cxxSettings: [
+                .headerSearchPath("Library/sqlite-amalgamation"),
+                .headerSearchPath("KeenClient"),
+                .headerSearchPath("KeenClientInternal"),
```
[the remark this answers is not in the document]
Comment (nacho4d) [reply Package.swift]: Thank you. I have moved KeenConstants to KeenClient folder 👍

**Review by tung-vu-td — APPROVED** [review]
Thanks @nacho4d!

**Review by nmpennypacker — APPROVED** [review]
Reviewed for data exfiltrations

**Pull request merged**